rs/boundary_node/anonymization/client/src/lib.rs

```diff
@@ -28,6 +83,32 @@ pub trait Register: Sync + Send {
     async fn register(&self, pubkey: &[u8]) -> Result<(), RegisterError>;
 }
 
+#[async_trait]
+impl<T: Register> Register for WithLogs<T> {
+    async fn register(&self, pubkey: &[u8]) -> Result<(), RegisterError> {
+        let start_time = Instant::now();
+
+        let out = self.0.register(pubkey).await;
+
+        let status = match &out {
+            Ok(_) => "ok",
+            Err(err) => match err {
+                RegisterError::Unauthorized => "unauthorized",
+                RegisterError::UnexpectedError(_) => "fail",
+            },
+        };
+
+        let duration = start_time.elapsed().as_secs_f64();
+
+        println!(
+            "action = 'register', status = {status}, duration = {duration}, error = {:?}",
+            out.as_ref().err()
+        );
```

Comment:
Shouldn't this write to some logger that is being passed in?

Reply:
coming soon™️